### ISSUE TITLE
Datasources: UID is invalid error when checking health of provisioned datasource

### ISSUE THREAD

**Issue by marefr** [grafana/grafana#68447]
### 
# What went wrong?

**What happened**:

  - Clicked Save & test of a provisioned datasource and received `UID is invalid` error from the check health request (`api/datasources/uid/<uid>/health)

**What did you expect to happen**:

  - No error

### 
# How do we reproduce it?

**Step 1**:

- Provision a datasource according to below
```yaml
apiVersion: 1

deleteDatasources:
  - name: Oracle 11 (Integration)

datasources:
  - name: Oracle 11 (Integration)
    uid: Oracle 11 (Integration)
    type: tempo
    access: proxy
    url: http://localhost:3200
    editable: true
    version: 1
```

**Step 2**:

- Browse the configuration of above provisioned datasource and click Save & test and the error message `UID is invalid` should be returned

### 
# What Grafana version are you using?

Grafana v10.1.0-55917pre (18aba61cbd)

### 
### Grafana Platform?

I use Grafana Cloud

### Is this a Regression?

None

### Anything else to add?

Seems to me like Grafana doesn't validate that datasource UID is valid when being provisioning and/or when create/update a datasource through the API. If that would have been in place this failure on check health couldn't   happen in the first place. Question is what do we do, if anything, with existing datasources having an invalid uid?

Here's the provisioning logic for generating UID if empty:
https://github.com/grafana/grafana/blob/9441692fe9953ca2ddb279029f77c710a4d69a2d/pkg/services/provisioning/datasources/types.go#L197-L227

Here's the default logic if UID is empty:
https://github.com/grafana/grafana/blob/9441692fe9953ca2ddb279029f77c710a4d69a2d/pkg/services/datasources/service/store.go#L241-L247

**Comment by marefr** [comment]
If making these changes and provisioning a datasource with invalid uid it would fail and basically Grafana would crash.

**Comment by marefr** [comment]
Decision: This problem has been there since v9.4.4 so we shouldn't rush to squeeze it into v10.
Lets deprecate for v10.1 or similar and log an error saying that the UID is invalid (deprecated) but still load it (issue to be created). Then do the breaking change for v11.

**Comment by tolzhabayev** [comment]
@marefr can you please update this issue or create a new one that describes in more details on what we want to do:

> Lets deprecate for v10.1 or similar and log an error saying that the UID is invalid (deprecated) but still load it (issue to be created). Then do the breaking change for v11.

**Comment by tolzhabayev** [comment]
Feels like this one is done via https://github.com/grafana/grafana/pull/68800 @marefr ?

**Comment by marefr** [comment]
Closed by #68800. Follow up issue #73892 created
